pyvista/plotting/plotting.py

```diff
@@ -398,35 +428,47 @@ def _to_pos(elevation, azimuth):
             z = np.cos(theta) * np.sin(phi)
             return x, y, z
 
+        renderers = [self.renderer] if only_active else self.renderers
+        for renderer in renderers:
+            renderer.remove_all_lights()
+
         # Inspired from Mayavi's version of Raymond Maple 3-lights illumination
-        lights = list(self.renderer.GetLights())
-        headlight = lights.pop(0)
-        headlight.SetSwitch(False)
-        for i in range(len(lights)):
-            if i < 3:
-                lights[i].SetSwitch(True)
-                lights[i].SetIntensity(1.0)
-                lights[i].SetColor(1.0, 1.0, 1.0)
-            else:
-                lights[i].SetSwitch(False)
-                lights[i].SetPosition(_to_pos(0.0, 0.0))
-                lights[i].SetIntensity(1.0)
-                lights[i].SetColor(1.0, 1.0, 1.0)
-
-        lights[0].SetPosition(_to_pos(45.0, 45.0))
-        lights[1].SetPosition(_to_pos(-30.0, -60.0))
-        lights[1].SetIntensity(0.6)
-        lights[2].SetPosition(_to_pos(-30.0, 60.0))
-        lights[2].SetIntensity(0.5)
-
-    def disable_3_lights(self):
-        """Disable 3-lights illumination."""
-        self.lighting = vtk.vtkLightKit()
-        # self.lighting.SetHeadLightWarmth(1.0)
-        # self.lighting.SetHeadLightWarmth(1.0)
-        for renderer in self.renderers:
-            renderer.RemoveAllLights()
-            self.lighting.AddLightsToRenderer(renderer)
+        intensities = [1, 0.6, 0.5]
+        all_angles = [(45.0, 45.0), (-30.0, -60.0), (-30.0, 60.0)]
+        for intensity, angles in zip(intensities, all_angles):
+            light = pyvista.Light(light_type='camera light')
+            light.intensity = intensity
+            light.position = _to_pos(*angles)
+            for renderer in renderers:
+                renderer.add_light(light)
```

Comment:
This is *way* cleaner.  I can really see the advantages of having a pythonic lighting class.